### PR TITLE
Utah at-home parent credit refundability is backwards

### LINKED ISSUE #3839: Utah at-home parent credit refundability is backwards
Think we need a `not` here:

https://github.com/PolicyEngine/policyengine-us/blob/72660fd4eefdaf8d033f9cc0129b8bcd2179b586/policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py#L32-L41

### PR DESCRIPTION
**Pull request by Mshuning**
Fixes #3839

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
+  changes:
+    added:
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:3]:
```suggestion
    fixed:
```

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+      - Utah at-home parent credit refundability update
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
      - Utah at-home parent credit refundability.
```

**Review by PavelMakarchuk — APPROVED** [review]

**Comment by MaxGhenis** [comment]
While you're here could you "modernize" this variable by adopting conventions like eligibility variables and add?

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: delete the refundable yaml file 

policyengine_us/parameters/gov/states/ut/tax/income/credits/at_home_parent/refundable.yaml

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibility.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        irs_employment_income: 1_000
+        self_employment_income: 0
+    households:
+      household:
+        members: [person1, person2]
+        state_code: UT
+  output: # 1 * 1
+    ut_at_home_parent_credit_eligibility: true
+
+- name: one parent qualifies, tax unit not qualify
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibility.yaml:18]:
```suggestion
- name: one parent qualifies, tax unit does not qualify
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibility.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
+- name: both qualify
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibility.yaml:1]:
```suggestion
- name: Both parents qualify
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibility.yaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        members: [person1, person2]
+        adjusted_gross_income: 80_000
+    households:
+      household:
+        members: [person1, person2]
+        state_code: UT
+  output: # 1 * 0
+    ut_at_home_parent_credit_eligibility: false
+
+- name: one parent not qualify, tax unit qualifies
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibility.yaml:39]:
```suggestion
- name: one parent does not qualify, tax unit qualifies
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibility.yaml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        irs_employment_income: 10_000
+        self_employment_income: 0
+    households:
+      household:
+        members: [person1, person2]
+        state_code: UT
+  output: # 0 * 1
+    ut_at_home_parent_credit_eligibility: false
+
+- name: both not qualify
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibility.yaml:56]:
```suggestion
- name: Both parents do not qualify
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibile.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+from policyengine_us.model_api import *
+
+
+class ut_at_home_parent_credit_eligibile(Variable):
+    value_type = bool
+    entity = TaxUnit
+    label = "Eligible Tax Unit for the Utah at-home parent credit"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibile.py:7]:
```suggestion
    label = "Eligible for the Utah at-home parent credit"
```

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibile.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
+        p = parameters(period).gov.states.ut.tax.income.credits.at_home_parent
+
+        one_parent_income_eligible = (
+            person("irs_employment_income", period)
+            + person("self_employment_income", period)
+        ) < p.parent_max_earnings
+        employment_income_eligible = (
+            tax_unit.sum(one_parent_income_eligible) > 0
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibile.py:20]: shouldn't this filter on head_or_spouse?

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibile.py
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibile.py:1]: typo in filename eligibile

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibile.py (hunk 8 above)
Comment (Mshuning) [reply policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibile.py:20]: how about create a new variable `is_parent`?

Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_eligibile.py]: i don't think that's necessary since we're limiting the multiplication to children and step-parents count, so heads and spouses will capture it. but could you add a legal code reference to the python file to ensure we adhere to it?

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
looking at the law and instructions, i think both parents can claim the credit. [chatgpt agreed](https://chat.openai.com/share/d6d99f81-13ab-4301-a7e1-a6cf2a0f1638).

please make that change and also note that both income limits are <= not <

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py [hunk 9]
```diff
@@ -20,4 +20,8 @@ def formula(tax_unit, period, parameters):
         p = parameters(period).gov.states.ut.tax.income.credits.at_home_parent
         count_children = tax_unit.sum((age < p.max_child_age) & is_dependent)
 
-        return p.amount * count_children
+        is_joint = tax_unit("tax_unit_is_joint", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py:23]: each parent has to be independently eligible

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_one_parent_income_eligible.py [hunk 10]
```diff
@@ -0,0 +1,23 @@
+from policyengine_us.model_api import *
+
+
+class ut_at_home_parent_credit_one_parent_income_eligible(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_one_parent_income_eligible.py:4]:
```suggestion
class ut_at_home_parent_credit_earned_income_eligible_person(Variable):
```

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_one_parent_income_eligible.py [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    value_type = bool
+    entity = Person
+    label = "Eligible for the Utah at-home parent credit income"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_one_parent_income_eligible.py:7]:
```suggestion
    label = "Eligible person for the Utah at-home parent credit income based on the earned income"
```

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_one_parent_income_eligible.py [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    defined_for = StateCode.UT
+    reference = (
+        "https://le.utah.gov/xcode/Title59/Chapter10/59-10-S1005.html",
+        "https://www.taxformfinder.org/forms/2021/2021-utah-tc-40-full-packet.pdf#page=23",
+    )
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.states.ut.tax.income.credits.at_home_parent
+
+        head_or_spouse_eligible = person("is_tax_unit_head_or_spouse", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_one_parent_income_eligible.py:18]:
```suggestion
        head_or_spouse = person("is_tax_unit_head_or_spouse", period)
```

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_one_parent_income_eligible.py [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+        one_parent_income_eligible = (
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_one_parent_income_eligible.py:19]:
```suggestion
        earned_income_eligible = (
```

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_one_parent_income_eligible.py [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+            person("earned_income", period) <= p.parent_max_earnings
+        )
+
+        return (head_or_spouse_eligible * one_parent_income_eligible) > 0
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_one_parent_income_eligible.py:23]:
```suggestion
        return earned_income_eligible & head_or_spouse
```

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_agi_eligible.py [hunk 15]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class ut_at_home_parent_credit_agi_eligible(Variable):
+    value_type = bool
+    entity = TaxUnit
+    label = "Eligible for the Utah at-home parent credit AGI"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_agi_eligible.py:7]:
```suggestion
    label = "Eligible for the Utah at-home parent credit based on adjusted gross income"
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_agi_eligible.yaml [hunk 16]
```diff
@@ -0,0 +1,15 @@
+- name: Not qualify
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_agi_eligible.yaml:1]:
```suggestion
- name: AGI above the limit
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_agi_eligible.yaml [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+  period: 2022
+  input:
+    adjusted_gross_income: 80_000
+    state_code: UT
+  output: 
+    ut_at_home_parent_credit_agi_eligible: false
+
+- name: Qualify
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_agi_eligible.yaml:9]:
```suggestion
- name: AGI below the limit
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_agi_eligible.yaml [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+  period: 2022
+  input:
+    adjusted_gross_income: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_agi_eligible.yaml:12]: make this an edge case - exactly $1 off of the threshold

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.yaml [hunk 19]
```diff
@@ -3,68 +3,99 @@
...
-        ut_ss_benefits_credit: 150
+        ut_at_home_parent_credit_agi_eligible: false
     households:
       household:
         members: [person1, person2, person3]
         state_code: UT
   output:
     ut_at_home_parent_credit: 0
+
+- name: UT at-home parent credit unit test 3
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.yaml:48]: describe what the vase does

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.yaml [hunk 20]
```diff
@@ -3,68 +3,99 @@
...
+        members: [person1, person2, person3]
+        ut_at_home_parent_credit_agi_eligible: true
+    households:
+      household:
+        members: [person1, person2, person3]
+        state_code: UT
+  output: # 100 * 2
+    ut_at_home_parent_credit: 200
+
+- name: UT at-home parent credit unit test 4
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.yaml:73]:
```suggestion
- name: describe what the test does
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.yaml [hunk 21]
```diff
@@ -1,70 +1,101 @@
...
       household:
-        members: [person1, person2, person3]
+        members: [person1, person2]
         state_code: UT
-  output:
+  output: # 100 * 1
     ut_at_home_parent_credit: 100
 
-- name: UT at-home parent credit unit test 2
+- name: Earned income eligible, agi not eligible
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.yaml:23]:
```suggestion
- name: Earned income eligible but not AGI eligible
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_earned_income_eligible_person.yaml [hunk 22]
```diff
@@ -0,0 +1,35 @@
+- name: Income does not qualify, parent
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_earned_income_eligible_person.yaml:1]:
```suggestion
- name: Income does not qualify, head or spouse
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_earned_income_eligible_person.yaml [hunk 23]
```diff
@@ -0,0 +1,35 @@
...
+  period: 2022
+  input:
+    earned_income: 3_001
+    is_tax_unit_head_or_spouse: true
+    state_code: UT
+  output: 
+    ut_at_home_parent_credit_earned_income_eligible_person: false
+
+- name: Income qualifies, parent
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_earned_income_eligible_person.yaml:10]:
```suggestion
- name: Income qualifies, head or spouse
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_earned_income_eligible_person.yaml [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+  period: 2022
+  input:
+    earned_income: 3_000
+    is_tax_unit_head_or_spouse: true
+    state_code: UT
+  output: 
+    ut_at_home_parent_credit_earned_income_eligible_person: true
+
+- name: Income qualifies, not parent
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_earned_income_eligible_person.yaml:19]:
```suggestion
- name: Income qualifies, not head or spouse
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_earned_income_eligible_person.yaml [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+  period: 2022
+  input:
+    earned_income: 3_000
+    is_tax_unit_head_or_spouse: false
+    state_code: UT
+  output: 
+    ut_at_home_parent_credit_earned_income_eligible_person: false
+
+- name: Income does not qualify, not parent
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_at_home_parent_credit_earned_income_eligible_person.yaml:28]:
```suggestion
- name: Income over the limit, not head or spouse
```

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py [hunk 26]
```diff
@@ -7,35 +7,22 @@ class ut_at_home_parent_credit(Variable):
...
         count_children = tax_unit.sum((age < p.max_child_age) & is_dependent)
-        parent_qualifies = (
-            person("irs_employment_income", period)
-            + person("self_employment_income", period)
-        ) < p.parent_max_earnings
-        one_parent_qualifies = tax_unit.sum(parent_qualifies) > 0
-        tax_unit_qualifies = (
-            tax_unit("adjusted_gross_income", period) < p.max_agi
+
+        one_parent_income_eligible = person(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py:23]:
```suggestion
        income_eligible_person = person(
```

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py [hunk 27]
```diff
@@ -7,35 +7,22 @@ class ut_at_home_parent_credit(Variable):
...
+            "ut_at_home_parent_credit_earned_income_eligible_person", period
         )
-        max_credit = (
-            p.amount
-            * count_children
-            * one_parent_qualifies
-            * tax_unit_qualifies
+        return tax_unit.sum(
+            p.amount * count_children * one_parent_income_eligible
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py:27]:
```suggestion
        individual_credit_amount = p.amount * count_children * one_parent_income_eligible
        return tax_unit.sum(
            individual_credit_amount
```

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py:27]: just to confirm, if two parents are eligible and they have 2 kids they get 4x the amount?

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py (hunk 27 above)
Comment (Mshuning) [reply policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py]: https://chat.openai.com/share/d6d99f81-13ab-4301-a7e1-a6cf2a0f1638

> (2) For a taxable year beginning on or after January 1, 2000, a claimant may claim on the claimant's individual income tax return a nonrefundable tax credit of $100 for each qualifying child if:
(a) the claimant or another claimant filing a joint individual income tax return with the claimant is an at-home parent; and 
(b) the adjusted gross income of all of the claimants filing the individual income tax return is less than or equal to $50,000.

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py [hunk 28]
```diff
@@ -7,35 +7,24 @@ class ut_at_home_parent_credit(Variable):
...
-                - tax_unit("ut_taxpayer_credit", period)
-                - tax_unit("ut_eitc", period)
-                - tax_unit("ut_retirement_credit", period)
-                - tax_unit("ut_ss_benefits_credit", period)
+
+        income_eligible_person = tax_unit.sum(
+            person(
+                "ut_at_home_parent_credit_earned_income_eligible_person",
+                period,
             )
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_at_home_parent_credit.py:27]:
```suggestion
        income_eligible_person = person(
                "ut_at_home_parent_credit_earned_income_eligible_person",
                period,
            )
        eligible_people = tax_unit.sum(income_eligible_person)
```

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]
